### PR TITLE
feat: added create feature command

### PR DESCRIPTION
**Pull request by jsalaber**
# Changes

- Added Create Feature command, can create with variables and SDK visibility as prompts

### CODE REVIEW

**Comment by jsalaber** [comment]
Current dependencies on/for this PR:
* main
  * **PR #160** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/cli/160" target="_blank">Graphite</a> 
    * **PR #166** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/cli/166" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/cli/166?utm_source=stack-comment).

File: src/commands/createCommand.ts [hunk 1]
```diff
@@ -1,7 +1,14 @@
 import { Flags } from '@oclif/core'
 import Base from './base'
 
+type Prompt = {
```
Comment (nsalamad) [inline src/commands/createCommand.ts:4]: this type is defined in `src/ui/prompts/types.ts` now so you can use that

File: src/commands/features/create.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    prompts = [keyPrompt, namePrompt, descriptionPrompt]
+
+    public async run(): Promise<void> {
+        const { flags } = await this.parse(CreateFeature)
+        const { headless, key, name, description, variables, variations, sdkVisibility } = flags
+        await this.requireProject()
+
+        if (headless && (!key || !name)) {
+            this.writer.showError('In headless mode, the key and name flags are required')
```
Comment (nsalamad) [inline src/commands/features/create.ts:29]: Laura mentioned on my PR that the 'In headless mode` part isn't needed

File: src/commands/features/create.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            return
+        }
+
+        const params = await this.populateParameters(CreateFeatureParams, this.prompts, {
+            key,
+            name,
+            description,
+            variables: variables ? JSON.parse(variables) : [],
```
Comment (nsalamad) [inline src/commands/features/create.ts:37]: What happens if the variables/variations string is malformed? Does this just throw an exception?

Comment (jsalaber) [reply src/commands/features/create.ts:37]: i haven't committed it yet but I'm gonna wrap most of this functionality in a try catch so we can properly catch errors

File: src/commands/targeting/get.ts [hunk 4]
```diff
@@ -42,7 +42,7 @@ export default class DetailedTargeting extends Base {
                     token: this.authToken,
                     projectKey: this.projectKey
                 }
-            )
+            ) as Record<string, any>
```
Comment (nsalamad) [inline src/commands/targeting/get.ts:45]: what's the purpose of this cast?

Comment (jsalaber) [reply src/commands/targeting/get.ts:45]: was to fix behaviour when I swapped an `any` to `string`, I think Laura fixed this so I can remove this

File: src/commands/features/create.ts (hunk 3 above)
Comment (nsalamad) [reply src/commands/features/create.ts]: ahh okay - nice!

File: src/api/features.ts [hunk 5]
```diff
@@ -1,12 +1,64 @@
...
+    variations?: CreateVariationParams[]
+
+    @IsOptional()
+    @Type(() => SDKVisibilityParams)
+    sdkVisibility?: SDKVisibilityParams
+}
+
+const FEATURE_URL = '/v1/projects/:project/features'
+
+const buildHeaders = (token: string) => ({
```
Comment (nsalamad) [inline src/api/features.ts:54]: might be better to move this into `src/api/common.ts` since it can be reused by other requests

**Review by nsalamad — APPROVED** [review]

**Pull request merged**